sdk/proto/engine.proto

```diff
@@ -33,7 +33,21 @@ enum LogSeverity {
 }
 
 message LogRequest {
-    LogSeverity severity = 1;   // the logging level of this message.
-    string message = 2;         // the contents of the logged message.
-    string urn = 3;             // the (optional) resource urn this log is associated with.
+    // the logging level of this message.
+    LogSeverity severity = 1;
+
+    // the contents of the logged message.
+    string message = 2;
+
+    // the (optional) resource urn this log is associated with.
+    string urn = 3;
+
+    // the (optional) stream id that a stream of log messages can be associated with. This allows
+    // clients to not have to buffer a large set of log messages that they all want to be
+    // conceptually connected.  Instead the messages can be sent as chunks (with the same stream id)
+    // and the end display can show the messages as they arrive, while still stitching them together
+    // into one total log message.
+    //
+    // 0/not-given means: do not associate with any stream.
+    int32 streamId = 4;
```

Comment:
Note: using 0 to mean 'not associated with any stream' is intentional.  This means that existing clients that call this RPC and do not provide this value, will still work and will simply have all their existing log messages assocaited with the non-sequently  default stream.